src/Plugin/Field/FieldFormatter/PlayerFormatter.php

```diff
@@ -244,23 +247,98 @@ protected function renderIframes(FieldItemListInterface $items, Player $player,
     /** @var \Drupal\media_mpx\Plugin\media\Source\Media $source_plugin */
     $source_plugin = $entity->getSource();
     foreach ($items as $delta => $item) {
-      try {
-        /** @var \Lullabot\Mpx\DataService\Media\Media $mpx_media */
-        $mpx_media = $source_plugin->getMpxObject($entity);
-      }
-      catch (TransferException $e) {
-        // If this media item is missing, continue on to the next element.
-        $this->mpxLogger->logException($e);
-        continue;
-      }
-      $url = new Url($source_plugin->getAccount(), $player, $mpx_media);
+      $element[$delta] = $this->buildWrapper($entity, $source_plugin, $player);
+    }
+  }
 
-      // @todo What cache contexts or tags do we set?
-      $element[$delta] = [
-        '#type' => 'media_mpx_iframe',
-        '#url' => (string) $url,
-      ];
+  /**
+   * Builds the render array for the wrapper.
+   *
+   * @param \Drupal\media\Entity\Media $entity
+   *   The media entity.
+   * @param \Drupal\media_mpx\Plugin\media\Source\Media $source_plugin
+   *   The MPX source plugin.
+   * @param \Lullabot\Mpx\DataService\Player\Player $player
+   *   The MPX player.
+   *
+   * @return array|null
+   *   The render array or null on an error.
+   */
+  private function buildWrapper(DrupalMedia $entity, Media $source_plugin, Player $player) {
+    try {
+      /** @var \Lullabot\Mpx\DataService\Media\Media $mpx_media */
+      $mpx_media = $source_plugin->getMpxObject($entity);
+    }
+    catch (TransferException $e) {
+      // If this media item is missing, continue on to the next element.
+      $this->mpxLogger->logException($e);
+      return NULL;
     }
+
+    $thumbnail_url = file_create_url($source_plugin->getMetadata($entity, 'thumbnail_uri'));
+
+    $element = [
+      '#type' => 'media_mpx_iframe_wrapper',
+      '#attributes' => [
+        'class' => [
+          'mpx-iframe-wrapper',
+        ],
+      ],
+      '#meta' => [
+        'name' => $entity->label(),
+        'thumbnailUrl' => $thumbnail_url,
+        'uploadDate' => $mpx_media->getAvailableDate()->format(DATE_ISO8601),
+      ],
+      '#content' => $this->buildPlayer($source_plugin, $player, $mpx_media),
+    ];
+    $this->addMediaFileDetails($element, $mpx_media);
+
+    return $element;
+  }
+
+  /**
+   * Adds the video specific details to schema.org meta data.
```

Comment:
`metadata` is usually one word.

I'm going to note that we pull from the first file.